integration-tests/src/test/java/org/apache/fineract/integrationtests/LoanTransactionChargebackTest.java

```diff
@@ -50,57 +55,263 @@
-                getLoanProductsProductResponse.getId().toString(), operationDate, "1000");
+        Set<GetLoansLoanIdTransactions> loanTransactions = getLoansLoanIdResponse.getTransactions();
+        assertNotNull(loanTransactions);
+        log.info("Loan Id {} with {} transactions", loanId, loanTransactions.size());
+        assertEquals(2, loanTransactions.size());
+        GetLoansLoanIdTransactions loanTransaction = loanTransactions.iterator().next();
+        log.info("Try to apply the Charge back over transaction Id {} with type {}", loanTransaction.getId(),
+                loanTransaction.getType().getCode());
+
+        applyChargebackTransaction(loanId, loanTransaction.getId().intValue(), amountVal, 0, responseSpecError);
+    }
+
+    @Test
+    public void applyLoanTransactionChargebackAfterMature() {
+        // Client and Loan account creation
+        final Integer loanId = createAccounts(45);
 
         GetLoansLoanIdResponse getLoansLoanIdResponse = loanTransactionHelper.getLoan(requestSpec, responseSpec, loanId);
         assertNotNull(getLoansLoanIdResponse);
 
         loanTransactionHelper.printRepaymentSchedule(getLoansLoanIdResponse);
+        GetLoansLoanIdRepaymentSchedule getLoanRepaymentSchedule = getLoansLoanIdResponse.getRepaymentSchedule();
+        log.info("Loan with {} periods", getLoanRepaymentSchedule.getPeriods().size());
+        assertEquals(2, getLoanRepaymentSchedule.getPeriods().size());
 
-        operationDate = Utils.dateFormatter.format(todaysDate);
-        Float amount = Float.valueOf("1100.0");
+        Float amount = Float.valueOf(amountVal);
         PostLoansLoanIdTransactionsResponse loanIdTransactionsResponse = loanTransactionHelper.makeLoanRepayment(operationDate, amount,
                 loanId);
         assertNotNull(loanIdTransactionsResponse);
         final Integer transactionId = loanIdTransactionsResponse.getResourceId();
 
-        List<GetPaymentTypesResponse> paymentTypeList = PaymentTypeHelper.getSystemPaymentType(requestSpec, responseSpec);
-        assertTrue(!paymentTypeList.isEmpty());
+        getLoansLoanIdResponse = loanTransactionHelper.getLoan(requestSpec, responseSpec, loanId);
+        assertNotNull(getLoansLoanIdResponse);
+        loanTransactionHelper.validateLoanStatus(getLoansLoanIdResponse, "loanStatusType.closed.obligations.met");
+
+        reviewLoanTransactionRelations(loanId, transactionId, 0);
+
+        final Integer chargebackTransactionId = applyChargebackTransaction(loanId, transactionId, "1000.00", 0, responseSpec);
+
+        reviewLoanTransactionRelations(loanId, chargebackTransactionId, 1);
+
+        getLoansLoanIdResponse = loanTransactionHelper.getLoan(requestSpec, responseSpec, loanId);
+        assertNotNull(getLoansLoanIdResponse);
+        loanTransactionHelper.validateLoanStatus(getLoansLoanIdResponse, "loanStatusType.active");
+
+        // N+1 Scenario
+        loanTransactionHelper.printRepaymentSchedule(getLoansLoanIdResponse);
```

Comment:
Please check whether the outstanding, principal , loan balance amount is correct after the chargeback

Reply:
Done!